dashboard/vendor/assets/javascripts/jquery.simulate.js

```diff
@@ -9,365 +9,400 @@
 
-      // Firefox won't trigger events if the window is inactive
-      // IE doesn't trigger events if we had to manually focus the body
-      if ( !triggered ) {
-        focusoutEvent = $.Event( "focusout" );
-        focusoutEvent.preventDefault();
-        element.trigger( focusoutEvent );
-        element.triggerHandler( "blur" );
-      }
-      element.unbind( "blur", trigger );
-    }, 1 );
+    return {
+      x: offset.left + elem.outerWidth() / 2 - document.scrollLeft(),
+      y: offset.top + elem.outerHeight() / 2 - document.scrollTop()
+    };
   }
-});
-
-
-
-/** complex events **/
-
-function findCenter( elem ) {
-  var offset,
-    document = $( elem.ownerDocument );
-  elem = $( elem );
-  offset = elem.offset();
-
-  return {
-    x: offset.left + elem.outerWidth() / 2 - document.scrollLeft(),
-    y: offset.top + elem.outerHeight() / 2 - document.scrollTop()
-  };
-}
 
-function findCorner( elem ) {
-  var offset,
-    document = $( elem.ownerDocument );
-  elem = $( elem );
-  offset = elem.offset();
+  function findCorner(elem) {
+    var offset,
+      document = $(elem.ownerDocument);
+    elem = $(elem);
+    offset = elem.offset();
 
-  return {
-    x: offset.left - document.scrollLeft(),
-    y: offset.top - document.scrollTop()
-  };
-}
+    return {
+      x: offset.left - document.scrollLeft(),
+      y: offset.top - document.scrollTop()
+    };
+  }
 
-var touchMappings = {};
-if ('ontouchstart' in document.documentElement) {
+  var touchMappings = {};
+  if ("ontouchstart" in document.documentElement) {
     touchMappings = {
-    mousedown: 'touchstart',
-    mousemove: 'touchmove',
-    mouseup: 'touchend'
-  };
-} else if (window.navigator.pointerEnabled) {  // IE 11+ support
+      mousedown: "touchstart",
+      mousemove: "touchmove",
+      mouseup: "touchend"
+    };
+  } else if (window.navigator.pointerEnabled) {
+    // IE 11+ support
     touchMappings = {
-    mousedown: 'pointerdown',
-    mousemove: 'pointermove',
-    mouseup: 'pointerup'
-  };
-} else if (window.navigator.msPointerEnabled) {  // IE 10 support
+      mousedown: "pointerdown",
+      mousemove: "pointermove",
+      mouseup: "pointerup"
+    };
+  } else if (window.navigator.msPointerEnabled) {
+    // IE 10 support
     touchMappings = {
-    mousedown: 'MSPointerDown',
-    mousemove: 'MSPointerMove',
-    mouseup: 'MSPointerUp'
-  };
-}
-
-$.extend( $.simulate.prototype, {
-  simulateDrag: function() {
-    if (window.Blockly && Blockly.version === 'Google' && Blockly.utils.global['PointerEvent']) {
-      touchMappings = {
-        mousedown: 'pointerdown',
-        mousemove: 'pointermove',
-        mouseup: 'pointerup'
+      mousedown: "MSPointerDown",
+      mousemove: "MSPointerMove",
+      mouseup: "MSPointerUp"
+    };
+  }
+
+  $.extend($.simulate.prototype, {
+    simulateDrag: function() {
+      if (
+        window.Blockly &&
+        Blockly.version === "Google" &&
+        window["PointerEvent"]
```

Comment:
@ebeastlake Here's the only actual code change. `Blockly.utils.global` was undefined. It's not actually clear to me why we we decided to use that 22 months ago. https://github.com/code-dot-org/code-dot-org/pulls?q=d815e6c73a49499a6b36303a3ad23f0e60fb0922

I determined that this worked by manually testing variations of `.simulate` commands in the browser console that matched what we needed for our tests that drag blocks.
https://codedotorg.atlassian.net/browse/SL-418

Reply:
See: https://github.com/google/blockly/blob/2257b1da19f562fe427606f5e8793ab76cc34f25/scripts/migration/renamings.json5#L1342

Reply:
Great find! I'll send a message to confirm, but let's try using `globalThis` instead of `window`. If I had seen this rename first, I would have tried it here.

Reply:
Never mind. Per our Slack conversation let's leave this as `window` for now. 
(I missed that you're not meant to use `globalThis` directly.)